### PR TITLE
[42898] Added possibility to filter by build statuses in config

### PR DESCRIPTION
**Pull request by Gandi24**
The idea is to create a config variable with this schema:

```
jenkins_filtering:
  RobotTest.git:
    - FAILURE
    - UNSTABLE
```

### CODE REVIEW

File: jaraco/pmxbot/http.py [hunk 1]
```diff
@@ -52,8 +52,20 @@ def default(self, channel=None):
...
+
+	def display_message(self, name, url, build, **kwargs):
+		"""
+		Decides if message should be displayed based on its status and config.
+		Mostly done to not display the SUCCESS statuses of periodic builds.
+		Options are SUCCESS, UNSTABLE, FAILED, ABORTED, and CYCLE.
+		"""
+		jenkins_filtering = pmxbot.config.get("jenkins_filtering", {})
+		job_filtering = jenkins_filtering.get(name, [])
+		if not job_filtering or build.get('status') in job_filtering:
```
Comment (jaraco) [inline jaraco/pmxbot/http.py]: Here, I would simply `return not job_filtering or build.get('status') in job_filtering`.

**Comment by Gandi24** [comment]
Adjusted the config:

```
Jenkins channels:
    Address.git: g
    pan:
    - panoptic
    - gryphon
    - cumulus
    RobotTest.git:
        channels: [qa, is, sandbox]
        statuses: [FAILURE, UNSTABLE]
```

**Comment by jaraco** [comment]
I'll get this out in a release, so as to avoid delays for production, but do please consider sending another PR with a test.

**Comment by jaraco** [comment]
Released as 3.10.

**Pull request merged**